relay-server/src/metrics.rs

```diff
@@ -0,0 +1,200 @@
+            RelayTimers::EventProcessingSerialization => "event_processing.serialization",
+            RelayTimers::EventWaitTime => "event.wait_time",
+            RelayTimers::EventProcessingTime => "event.processing_time",
+            RelayTimers::EventTotalTime => "event.total_time",
+            RelayTimers::ProjectStateEvictionDuration => "project_state.eviction.duration",
+            RelayTimers::ProjectStateRequestDuration => "project_state.request.duration",
+            RelayTimers::ProjectIdRequestDuration => "project_id.request.duration",
+            RelayTimers::RequestsDuration => "requests.duration",
+        }
+    }
+}
+
+/// Counter metrics used by Relay
+pub enum RelayCounters {
+    /// Number of events accepted in the current time slot. This represents events that
+    /// have successfully passed rate limits, filters and have been successfully handled.
+    EventAccepted,
+    /// Number of events rejected in the current time slot. This includes events being rejected
+    /// because they are malformed or any other error during processing (including filtered
+    /// events, discarded events and rate limited events).
+    EventRejected,
+    /// Represents a group of counters, implemented with using tags. The following tags are
+    /// present for each event outcome: `outcome` which is an `EventOutcome` enum and reason
+    /// which is the reason string for all outcomes that are not `Accepted`.
+    EventOutcomes,
+    /// Counts the number of times a project state lookup is done. This includes requests
+    /// for projects that are cached and requests for projects that are not yet cached.
+    /// All requests that return a  `EventAction::Accept` i.e. are not rate limited (on
+    /// the fast path) or are discarded because we know the project is disabled or invalid
+    /// will be counted.
+    ProjectStateGet,
+    /// Counts the number of project state http requests. Note that a project state HTTP request
+    /// typically contains a number of projects (the project state requests are batched).
+    ProjectStateRequest,
+    /// Counts the number of times a request for a project is already present, this effectively
+    /// represents the fraction of ProjectStateGet that will **not** result in a ProjectState
+    /// request.
+    ProjectCacheHit,
+    /// Counts the number of times a request for a project is not already present.
+    /// `ProjectStateGet` = `ProjectCacheMiss` + `ProjectCacheHit`.
+    /// Requests that are generating a cache hit will be queued and batched and eventually will
+    /// generate a `ProjectStateRequest`.
+    ProjectCacheMiss,
+    /// Counts the number of requests for the  ProjectId (the timing is tracked
+    /// by `ProjectIdRequestDuration`). Note that only projects with the id not already fetched
+    /// are counted. Once the ProjectId is successfully cached if will be retained indefinitely.
+    ProjectIdRequest,
```

Comment:
It's important to note here that this is part of the legacy store endpoint. Regular requests obtain their project id from the URL.